pkg/account/service.go

```diff
@@ -133,6 +138,68 @@ func (a *Service) Update(ID string, data *Account) (*Account, error) {
 	return account, nil
 }
 
+// Create creates a new account using the data provided. Returns the account record
+func (a *Service) Create(data *Account) (*Account, error) {
+	// Validate the incoming record doesn't have unneeded fields
+	err := validation.ValidateStruct(data,
+		// This may be considered double validation but we are going to need the ID
+		// so need to make sure its set
+		validation.Field(&data.ID, validateID...),
+		validation.Field(&data.AdminRoleArn, validateAdminRoleArn...),
+		validation.Field(&data.LastModifiedOn, validation.By(isNil)),
+		validation.Field(&data.Status, validation.By(isNil)),
+		validation.Field(&data.CreatedOn, validation.By(isNil)),
+		validation.Field(&data.PrincipalRoleArn, validation.By(isNil)),
+		validation.Field(&data.PrincipalPolicyHash, validation.By(isNil)),
+	)
+	if err != nil {
+		return nil, errors.NewValidation("account", err)
+	}
+
+	// Check if account already exists
+	existingAccount, err := a.Get(*data.ID)
+	if existingAccount != nil {
+		return nil, errors.NewAlreadyExists("account", *data.ID)
+	}
+	if err != nil {
+		if !errors.Is(err, errors.NewNotFound("account", *data.ID)) {
+			return nil, err
+		}
+	}
+
+	new, err := NewAccount(NewAccountInput{
+		ID:                *data.ID,
+		AdminRoleArn:      *data.AdminRoleArn,
+		Metadata:          data.Metadata,
+		PrincipalRoleName: a.principalRoleName,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.managerSvc.UpsertPrincipalAccess(new)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.Save(new)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.eventSvc.AccountCreate(new)
```

Comment:
I like this setup, but I think the difference of intention between `eventSvc.AccountCreate()` and `eventSvc.AccountReset()` is a little fuzzy.

For `AccountCreate`, we're saying that we _did_ create an account, for anyone who cares
For `AccountReset`, we're saying that we _want to_ reset the account.

It's just a naming change, but I think it would be more clear as something `AccountCreated`(passive) / `ResetAccount` (imperative)

Or I've even setup event topics like `DidCreateAccount` / `PleaseResetAccount`, to be extra explicit.

Reply:
would also be worth an inline command on line 195, like "this adds the account to the reset queue, to trigger nuke to run on the account"

Reply:
Yea let me see if I can find some good naming for this.  Good call.